format/spec.md

```diff
@@ -598,6 +702,14 @@ Notes:
 1. Lower and upper bounds are serialized to bytes using the single-object serialization in Appendix D. The type of used to encode the value is the type of the partition field data.
 2. If -0.0 is a value of the partition field, the `lower_bound` must not be +0.0, and if +0.0 is a value of the partition field, the `upper_bound` must not be -0.0.
 
+#### First Row ID Assignment
+
+Row ID inheritance is used when row lineage is enabled. When not enabled, a manifest's `first_row_id` must always be set to `null`. The rest of this section applies when row lineage is enabled.
```

Comment:
@rdblue also has some thoughts on this. I don't have a problem with enabling and disabling and having slightly odd behavior when that happens. I think it's a pretty unlikely situation to have it flip back and forth, for users who do that they can expect some odd events.

Reply:
I don't think that we should allow disabling it. That would create strange situations and we would not be able to trust the metadata. There may be a way to handle this, but I don't want to block the initial feature taking the time to design it.

Reply:
I would assume we would treat any commits in the gap the same as we would treat writes before tracking was on.

Reply:
A related question: if we revert the table to a snapshot before enabling the row lineage, should we disable row lineage? If not, what about `next_row_id`?

Reply:
Can we be more explicit in the spec about disabling row lineage is not allowed at this moment? I think the engine/catalog should guide it so that users won't accidentally disable it.

Reply:
A table snapshot reversion is the same as resetting current-snapshot. In this case we don't change next_row_id and we would not disable row_lineage. Turning on `row_lineage` is not a snapshot operation. I'll write that explicitly.